src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/openapi/TimestampingServicesApiController.java

```diff
@@ -62,9 +65,42 @@ public TimestampingServicesApiController(TimestampingServiceService timestamping
 
     @Override
     @PreAuthorize("hasAuthority('VIEW_TSPS')")
-    public ResponseEntity<List<TimestampingService>> getTimestampingServices() {
-        Collection<String> urls = timestampingServiceService.getTimestampingServices();
-        List<TimestampingService> timestampingServices = timestampingServiceConverter.convert(urls);
+    public ResponseEntity<List<TimestampingService>> getTimestampingServices(Boolean showConfigured) {
+        List<TimestampingService> timestampingServices;
+        // If show configured is true, return configured timestamping services from serverconf db.
+        // Otherwise return approved timestamping services from global conf.
+        if (showConfigured) {
+            List<TspType> tsps = timestampingServiceService.getConfiguredTimestampingServices();
+            timestampingServices = timestampingServiceConverter.convert(tsps);
+        } else {
+            Collection<String> urls = timestampingServiceService.getApprovedTimestampingServices();
+            timestampingServices = timestampingServiceConverter.convert(urls);
+        }
         return new ResponseEntity<>(timestampingServices, HttpStatus.OK);
     }
+
+    @Override
+    @PreAuthorize("hasAuthority('ADD_TSP')")
+    public ResponseEntity<TimestampingService> addTimestampingService(TimestampingService timestampingServiceToAdd) {
+        try {
+            timestampingServiceService.addConfiguredTimestampingService(timestampingServiceToAdd);
+        } catch (TimestampingServiceService.DuplicateConfiguredTimestampingServiceException e) {
+            throw new ConflictException(e);
+        } catch (TimestampingServiceNotFoundException e) {
+            throw new ResourceNotFoundException(e);
+        }
+        return new ResponseEntity<>(timestampingServiceToAdd, HttpStatus.CREATED);
```

Comment:
I ended up leaving this as-is - not including Location header since we do not have an endpoint for a single timestamping entity.

Reply:
With a fresh mind this (b) is indeed the better option since it is better to not provide _wrong_ Location header if we cannot provide a correct one.  Discussed same thing with @TJaakkola today, about Endpoint delete, since we don't have a single item getter for that either.